azurerm/resource_arm_automation_schedule.go

```diff
@@ -282,3 +377,81 @@ func resourceArmAutomationScheduleDelete(d *schema.ResourceData, meta interface{
 
 	return nil
 }
+
+func expandArmAutomationScheduleAdvanced(d *schema.ResourceData, isUpdate bool) (*automation.AdvancedSchedule, error) {
+
+	expandedAdvancedSchedule := automation.AdvancedSchedule{}
+
+	// If frequency is set to `Month` the `week_days` array cannot be set (even empty), otherwise the API returns an error.
+	// During update it can be set and it will not return an error. Workaround for the APIs behavior
+	if v, ok := d.GetOk("week_days"); ok {
+		weekDays := v.(*schema.Set).List()
+		expandedWeekDays := make([]string, len(weekDays))
+		for i := range weekDays {
+			expandedWeekDays[i] = weekDays[i].(string)
+		}
+		expandedAdvancedSchedule.WeekDays = &expandedWeekDays
+	} else if isUpdate {
+		expandedAdvancedSchedule.WeekDays = &[]string{}
+	}
+
+	// Same as above with `week_days`
+	if v, ok := d.GetOk("month_days"); ok {
+		monthDays := v.(*schema.Set).List()
+		expandedMonthDays := make([]int32, len(monthDays))
+		for i := range monthDays {
+			expandedMonthDays[i] = int32(monthDays[i].(int))
+		}
+		expandedAdvancedSchedule.MonthDays = &expandedMonthDays
+	} else if isUpdate {
+		expandedAdvancedSchedule.MonthDays = &[]int32{}
+	}
+
+	monthlyOccurrences := d.Get("monthly_occurrence").([]interface{})
+	expandedMonthlyOccurrences := make([]automation.AdvancedScheduleMonthlyOccurrence, len(monthlyOccurrences))
+	for i := range monthlyOccurrences {
+		m := monthlyOccurrences[i].(map[string]interface{})
+		occurrence := int32(m["occurrence"].(int))
+
+		expandedMonthlyOccurrences[i] = automation.AdvancedScheduleMonthlyOccurrence{
+			Occurrence: &occurrence,
+			Day:        automation.ScheduleDay(m["day"].(string)),
+		}
+	}
+	expandedAdvancedSchedule.MonthlyOccurrences = &expandedMonthlyOccurrences
+
+	return &expandedAdvancedSchedule, nil
+}
+
+func flattenArmAutomationScheduleAdvancedWeekDays(v *automation.AdvancedSchedule) *schema.Set {
+	flattenedWeekDays := schema.NewSet(set.HashStringIgnoreCase, []interface{}{})
+	if v.WeekDays != nil {
```

Comment:
I believe this could be written as:
```go
    if weekDays := v.WeekDays; weekdays != nil {
        for i, v := range *weekDays {
			flattenedWeekDays.Add(v)
		}
	}
```